### PR TITLE
[FEAT] Entry phase

### PR DESCRIPTION
**Pull request by yoavabadi**
New Feature - Entry Phase!
Now, we can allow the user to add an `entry_phase` with options, so we can resend the options to to the operation from the entry phase.

This can be extremely useful for re-sending events from a fail-queue feature.

### CODE REVIEW

File: operation/operation.py [hunk 1]
```diff
@@ -15,7 +16,11 @@ def __init__(self, options={}):
 
     def run(self):
         try:
-            for phase in self.phases():
+            phases = self.phases()
+            if self._entry_phase:
+                phases = phases[phases.index(self._entry_phase):]
```
Comment (segalzeyalz) [inline operation/operation.py]:
```python
start_phase = phases.index(self._entry_phase)
phases = phases[start_phase:]
```

File: operation/operation.py [hunk 2]
```diff
@@ -31,7 +36,7 @@ def run(self):
         return self
 
     def phases(self):
```
Comment (segalzeyalz) [inline operation/operation.py:41]: Should be abstract method:
`    @abstractmethod
`

File: operation/operation.py [hunk 3]
```diff
@@ -3,9 +3,10 @@
 
 
 class Operation:
```
Comment (segalzeyalz) [inline operation/operation.py]: `Operation(ABC)`

File: tests/test_operation.py [hunk 4]
```diff
@@ -28,8 +28,8 @@ def phase_test_fail(self):
             self.fail_phase()
```
Comment (segalzeyalz) [inline tests/test_operation.py:30]: Add test for this logic

Comment (yoavabadi) [reply tests/test_operation.py:30]: there is one

**Pull request merged**